packages/expo-router/src/typed-routes/types.ts

```diff
@@ -0,0 +1,221 @@
+export type StaticRoutes = ExpoRouter.__routes extends { StaticRoutes: string }
+  ? // @ts-expect-error
+    ExpoRouter.__routes['StaticRoutes']
+  : string;
+export type DynamicRoutes<T extends string> =
+  ExpoRouter.__routes<T> extends {
+    DynamicRoutes: any;
+  }
+    ? // @ts-expect-error
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      T extends ExpoRouter.__routes<infer _>['DynamicRoutes']
+      ? T
+      : never
+    : string;
+export type DynamicRouteTemplate = ExpoRouter.__routes extends { DynamicRouteTemplate: string }
+  ? // @ts-expect-error
+    ExpoRouter.__routes['DynamicRouteTemplate']
+  : string;
```

Comment:
My only concern here are `@ts-expect-error` annotations. if we don't manage this without then there's a possibility that, if the annotations aren't preserved in the declaration typings files, `skipLibCheck: false` users will see these errors, if the types weren't generated.

What error are we expecting to ignore here given the exact match?

Reply:
That seems to work fine 👍

Reply:
yea, constraining it with an even more exact interface via `extends` should also work. It just sometimes risks that the inference upcasts, but we could mitigate that with a variance annotation or a more explicit algebraic check.

Either way, `Record<string, unknown>` works just fine too of course 💯 Since the types aren't accessible / user-editable directly, I don't think we have to bother with the above